core/trino-main/src/main/java/io/trino/sql/planner/EqualityInference.java

```diff
@@ -142,14 +142,14 @@ public EqualityPartition generateEqualitiesPartitionedBy(Set<Symbol> scope)
                 }
             });
             // Compile the equality expressions on each side of the scope
-            Expression matchingCanonical = getCanonical(scopeExpressions);
+            Expression matchingCanonical = getCanonical(scopeExpressions.stream());
```

Comment:
Are there any calls to `getCanonical` where you don't need to use `.steram()`? If no, then we can drop this change

Reply:
We do, in getScopedCanonical().